chalicelib/checks/wrangler_checks.py

```diff
@@ -1021,3 +1021,77 @@ def add_grouped_with_file_relation(connection, **kwargs):
         action.status = 'DONE'
     action.output = action_logs
     return action
+
+@check_function(item_type=['VariantSample'])
+def core_project_status(connection, **kwargs):
+    """
+    Ensure CGAP Core projects have their objects shared.
+
+    Default behavior is to check only VariantSample objects, but defining 
+    'item_type' in check_setup.json will override the default and check status
+    for all objects defined there. 
+    """
+
+    check = CheckResult(connection, 'core_project_status')
+    item_type = kwargs.get('item_type')
+    full_output = {}
+    for item in item_type:
+        search_query = ('search/?project.display_title=CGAP+Core'
+                        '&type=' + item +
+                        '&status!=shared'
+                        '&frame=object&field=uuid')
+        not_shared = ff_utils.search_metadata(search_query,
+                                              key=connection.ff_keys)
+        if not_shared:
+            not_shared_uuids = []
+            for item_object in not_shared:
+                not_shared_uuids.append(item_object['uuid'])
+            full_output[item] = not_shared_uuids
+
+    if full_output:
+        check.status = 'WARN'
+        check.summary = 'Some CGAP Core items are not shared'
+        check.description = ('{} CGAP Core items do not have shared'
+                             'status'.format(sum([len(x) for x in 
+                                                  full_output.values()])))
+        brief_output = {key: len(value) for key, value in full_output.items()}
+        check.brief_output = brief_output
+        check.full_output = full_output
+        check.allow_action = True
+        check.action = 'share_core_project'
+    else:
+        check.status = 'PASS'
+        check.summary = check.description = ('All CGAP Core items are shared:'
+                                             '{}'.format(item_type))
```

Comment:
if the item_type list is long, this may not fit on the summary line in the UI - the summary could just be "All CGAP Core items are shared" and the description could add the itemtypes.